backend/model/users.py

```diff
@@ -57,10 +57,21 @@ def delete_from_db(self):
         self.state = False
         db.session.commit()
 
-    def update_from_db(self, password):
+    def update_password_from_db(self, password):
         self.hash_password(password)
         db.session.commit()
 
+    def update_from_db(self, data):
+        if 0 < self.query.filter_by(username=self.username and id != data['id'], state=True).count():
```

Comment:
Codi molt similar al del metode `save_to_db(self)`, es podria intentar unificar en un metode.
Tambe la part de la comprovacio `username=self.username` que s'esta fent es erronia perque no estas comprovant si el nou username entra amb conflicte.
```suggestion python
        if 0 < self.query.filter_by(username=data['username'] and id != self.id, state=True).count():
```